packages/react-sdk/src/embed-template.ts

```diff
@@ -104,10 +110,34 @@ const createInstancesFromTemplate = (
       // populate props
       if (item.props) {
         for (const prop of item.props) {
+          const propId = nanoid();
+          if (prop.dataSourceReference === undefined) {
+            props.push({ id: propId, instanceId, ...prop });
+            continue;
+          }
+          let dataSource = dataSourceByReference.get(prop.dataSourceReference);
+          if (dataSource === undefined) {
+            const id = nanoid();
+            const name = prop.dataSourceReference;
+            if (prop.type === "boolean") {
+              const { type, value: defaultValue } = prop;
+              dataSource = { id, name, type, defaultValue };
+            } else if (prop.type === "string") {
+              const { type, value: defaultValue } = prop;
+              dataSource = { id, name, type, defaultValue };
+            } else {
```

Comment:
same here regarding same code, assuming you wrote twice to satisfy ts but there might be another way so we don't end up with 10 of those later